### ISSUE TITLE
Q: callstack intended behaviour 

### ISSUE THREAD

**Issue by stefanofornari** [beanshell/beanshell#514]
Hello,
I wrote a simple command that prints the callstack when invoked:

```
public class printCallStack {
    public static void invoke(Interpreter i, CallStack callstack) {
        System.out.println(callstack);
    }
}
```
If I run:

```
foo() { printCallStack(); }
foo();
```

the result is:

```
CallStack:
	NameSpace: test (bsh.NameSpace@5e25a92e) (method) 
	NameSpace: global (bsh.NameSpace@70beb599)
```

Which seems fine.

if I write a script foo.bsh with the simple content:

`
printCallStack();`

And then I source it, the result is like the below:

```
source("foo.bsh");
CallStack:
	NameSpace: global (bsh.NameSpace@70beb599)
```

This does not sound right... I understand source() runs the script in the global namespace, but is it really the intended behaviour that it also does not appear in the callstack?

**Comment by stefanofornari** [comment]
@nickl- 

I would have expected something like:

```
source("foo.bsh");
CallStack:
	source (bsh.NameSpace@5e25a92e) (method) 
	NameSpace: global (bsh.NameSpace@70beb599)
```
but if the callstack is fundamentally a stack of the namespace something runs with, maybe there is not another way (because source must run the code like if called in the current namespace). But in this case we loose the information that source() has been invoked.

**Comment by nickl-** [comment]
Incorrect, if you look at the `source` command you'll see what happens. 

Also I think you imagine `CallStack` being something which it is not.

It is only a convenient way to pass the namespace to the AST and does not aid call tracking or stack tracing whatsoever.
